contracts/src/v0.8/test/HeartbeatRequester.t.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import 'forge-std/Test.sol';
+import {HeartbeatRequester} from "../HeartbeatRequester.sol";
+import '../mocks/MockAggregator.sol';
+import '../mocks/MockAggregatorProxy.sol';
+import '../mocks/MockOffchainAggregator.sol';
+
+    // from contracts directory,
+    // forge test --match-path src/v0.8/test/HeartbeatRequester.t.sol
+
+contract HeartbeatRequesterTest is Test {
+    HeartbeatRequester public heartbeatRequester;
+    MockAggregator public aggregator;
+    IAggregatorProxy public aggregatorProxy;
+
+    address internal constant USER = address(2);
+
+    event HeartbeatPermitted(address indexed permittedCaller, address newProxy, address oldProxy);
+    event HeartbeatRemoved(address indexed permittedCaller, address removedProxy);
+    error HeartbeatNotPermitted();
+
+    function setUp() public {
+        heartbeatRequester = new HeartbeatRequester();
+        aggregator = new MockAggregator();
+        aggregatorProxy = IAggregatorProxy(new MockAggregatorProxy(address(aggregator)));
+    }
+
+    function test_permitHeartbeat() public {
```

Comment:
add a test case for setting two proxies for the same address so we can test `oldProxy` field